sphinx/util/display.py

```diff
@@ -62,27 +62,29 @@ class SkipProgressMessage(Exception):
 
 
 class progress_message:
-    def __init__(self, message: str) -> None:
+    def __init__(self, message: str, nonl: bool = True) -> None:
```

Comment:
```suggestion
    def __init__(self, message: str, *, nonl: bool = True) -> None:
```

Just to keep the same logic as for `log(...)`.